### PR TITLE
DP-7584: Adds the sidebar-heading atom to storybook.

### PR DESCRIPTION
**Pull request by smurrayatwork**
**Description:** 
This adds the sidebar heading atom to storybook, as well as support for adding notes support. The sample data and markdown help text have been imported over as well.

**Jira:** 
https://jira.state.ma.us/browse/DP-7584

**To Test:**
- [ ] Pull the branch and run 'npm run storybook'.
- [ ] Browse to http://localhost:6006/ (or the port that you chose instead) and verify there's now an Atoms link on the left side. Click on it, then the link for Headings, then click on SidebarHeading. Verify that the text for 'Key Agencies' displays within an h2 tag with the class 'ma__sidebar-heading'. 
- [ ] Change the value for level in the SidebarHeading.json file located in the src/components/atoms/headings/SidebarHeading directory to be 1. Save the file and browse back to the sidebar heading from the previous step. Inspect the 'Key Agencies' text to verify that the h2 has changed to a h1 tag.

**Post Deploy Steps:**
N/A

**Relevant Screenshots/GIFs:**
screen shot 2018-01-25 at 4 05 52 pm

**Additional Notes:**
N/A
---

[Peer Review Checklist](docs/Peer%20review%20checklist.md)

### CODE REVIEW

File: src/components/atoms/headings/SidebarHeading/SidebarHeading.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const SidebarHeadingDynamicLevel = `h${sidebarHeading.level || 2}`;
+  return(
+    <SidebarHeadingDynamicLevel className="ma__sidebar-heading">
+      { sidebarHeading.title }
+    </SidebarHeadingDynamicLevel>
+  );
+};
+
+SidebarHeading.propTypes = {
+  data: PropTypes.node
```
Comment (mrjmd) [inline src/components/atoms/headings/SidebarHeading/SidebarHeading.js]: I think we can extend the propTypes to provide more checks inside the data object. See if you can test out using objectOf() and shape() inside here, for an example see: https://github.com/jsdrupal/drupal-react-components/blob/master/src/components/table/simple/index.js

Comment (smurrayatwork) [reply src/components/atoms/headings/SidebarHeading/SidebarHeading.js]: I did actually try shape before I did this, but eslint was throwing errors and I had to do this instead. I'll give it another shot, as I don't remember the error.

Comment (smurrayatwork) [reply src/components/atoms/headings/SidebarHeading/SidebarHeading.js]: Disregard; I was thinking of another PR.

**Comment by smurrayatwork** [comment]
@mrjmd This should be good to go.

File: src/components/atoms/headings/SidebarHeading/SidebarHeading.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const SidebarHeadingDynamicLevel = `h${sidebarHeading.level || 2}`;
+  return(
+    <SidebarHeadingDynamicLevel className="ma__sidebar-heading">
+      { sidebarHeading.title }
+    </SidebarHeadingDynamicLevel>
+  );
+};
+
+SidebarHeading.propTypes = {
+  data: PropTypes.shape({
```
Comment (mrjmd) [inline src/components/atoms/headings/SidebarHeading/SidebarHeading.js]: Based on @avrilpearl's comment on #10, nesting props under a `data` object may need to be revisited. I'll pass this internally for now but likely will need to await further discussion.

**Comment by smurrayatwork** [comment]
@mrjmd I moved away from using the data variable. This is ready for review.

**Pull request merged**